roles/ceph-mon/tasks/docker/main.yml

```diff
@@ -94,11 +94,20 @@
     - not containerized_deployment_with_kv
 
 - block:
+  - name: set_fact delegated_node
+    set_fact:
+      delegated_node: "{{ groups[mon_group_name] | first if rolling_update else groups[mon_group_name] | last }}"
+
+  - name: set_fact docker_exec_cmd_mgr_keyring
+    set_fact:
+      docker_exec_cmd_mgr_keyring: "{{ 'docker exec ceph-mon-' + hostvars[groups[mon_group_name]|first]['ansible_hostname'] if rolling_update else docker_exec_cmd }}"
+
   - name: create ceph mgr keyring(s) when mon is containerized
-    command: "{{ docker_exec_cmd }} ceph --cluster {{ cluster }} auth get-or-create mgr.{{ hostvars[item]['ansible_hostname'] }} mon 'allow profile mgr' osd 'allow *' mds 'allow *' -o /etc/ceph/{{ cluster }}.mgr.{{ hostvars[item]['ansible_hostname'] }}.keyring"
+    command: "{{ docker_exec_cmd_mgr_keyring }} ceph --cluster {{ cluster }} auth get-or-create mgr.{{ hostvars[item]['ansible_hostname'] }} mon 'allow profile mgr' osd 'allow *' mds 'allow *' -o /etc/ceph/{{ cluster }}.mgr.{{ hostvars[item]['ansible_hostname'] }}.keyring"
```

Comment:
We must use the following option `-m` to specify the first mon.

Reply:
Basically, this command still needs to run on the last monitor BUT must specify the first mon of the list.

Reply:
thanks maggle!